network/p2p/p2pconf/gossipsub_rpc_inspectors.go

```diff
@@ -17,102 +13,25 @@ type GossipSubRPCInspectorsConfig struct {
+	ControlMessageMaxSampleSize int `validate:"gte=1000" mapstructure:"gossipsub-rpc-control-message-max-sample-size"`
 }
 
-// AllCtrlMsgValidationConfig returns all control message validation configs in a list.
-func (conf *GossipSubRPCValidationInspectorConfigs) AllCtrlMsgValidationConfig() CtrlMsgValidationConfigs {
-	return CtrlMsgValidationConfigs{&CtrlMsgValidationConfig{
-		ControlMsg:      p2pmsg.CtrlMsgGraft,
-		HardThreshold:   conf.GraftLimits.HardThreshold,
-		SafetyThreshold: conf.GraftLimits.SafetyThreshold,
-		RateLimit:       conf.GraftLimits.RateLimit,
-	}, &CtrlMsgValidationConfig{
-		ControlMsg:      p2pmsg.CtrlMsgPrune,
-		HardThreshold:   conf.PruneLimits.HardThreshold,
-		SafetyThreshold: conf.PruneLimits.SafetyThreshold,
-		RateLimit:       conf.PruneLimits.RateLimit,
-	}, &CtrlMsgValidationConfig{
-		ControlMsg:      p2pmsg.CtrlMsgIHave,
-		HardThreshold:   conf.IHaveLimits.HardThreshold,
-		SafetyThreshold: conf.IHaveLimits.SafetyThreshold,
-		RateLimit:       conf.IHaveLimits.RateLimit,
-	}}
-}
-
-// CtrlMsgValidationConfigs list of *CtrlMsgValidationConfig
-type CtrlMsgValidationConfigs []*CtrlMsgValidationConfig
-
-// CtrlMsgValidationConfig configuration values for upper, lower threshold and rate limit.
-type CtrlMsgValidationConfig struct {
-	// ControlMsg the type of RPC control message.
-	ControlMsg p2pmsg.ControlMessageType
-	// HardThreshold specifies the hard limit for the size of an RPC control message.
-	// While it is generally expected that RPC messages with a size greater than HardThreshold should be dropped,
-	// there are exceptions. For instance, if the message is an 'iHave', blocking processing is performed
-	// on a sample of the control message rather than dropping it.
-	HardThreshold uint64 `mapstructure:"hard-threshold"`
-	// SafetyThreshold specifies the lower limit for the size of the RPC control message, it is safe to skip validation for any RPC messages
-	// with a size < SafetyThreshold. These messages will be processed as soon as possible.
-	SafetyThreshold uint64 `mapstructure:"safety-threshold"`
-	// RateLimit number of allowed messages per second, use 0 to disable rate limiting.
-	RateLimit int `mapstructure:"rate-limit"`
+// IWantRPCInspectionConfig validation configuration for iWANT RPC control messages.
+type IWantRPCInspectionConfig struct {
+	// MaxSampleSize max inspection sample size to use.
+	MaxSampleSize uint `validate:"gt=0" mapstructure:"gossipsub-rpc-iwant-max-sample-size"`
+	// CacheMissThreshold the threshold of missing corresponding iHave messages for iWant messages received before an invalid control message notification is disseminated.
+	CacheMissThreshold float64 `validate:"gt=0" mapstructure:"gossipsub-rpc-iwant-cache-miss-threshold"`
+	// DuplicateMsgIDThreshold maximum allowed duplicate message IDs in a single iWant control message.
+	DuplicateMsgIDThreshold float64 `validate:"gt=0" mapstructure:"gossipsub-rpc-iwant-duplicate-message-id-threshold"`
```

Comment:
For each of the above parameters, please provide a brief description to outline the behavior when exceeding the `max-sample-size`, as well as the consequences when the cache miss and duplicate message ID thresholds are reached

Reply:
https://github.com/onflow/flow-go/pull/4642/commits/9ec19ebf4679b95a674e18f5ad014fa95e19274a